src/main/com/marginallyclever/makelangelo/ScaleTurtlePanel.java

```diff
@@ -226,12 +217,9 @@ public static void main(String[] args) {
 		CommandLineOptions.setFromMain(args);
 		Translator.start();
 
-		try {
-			JFrame frame = new JFrame(ScaleTurtlePanel.class.getSimpleName());
-			frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
-			runAsDialog(frame,new Turtle());
-		} catch (Exception e) {
-			JOptionPane.showMessageDialog(null,e.getMessage(),"Error",JOptionPane.ERROR_MESSAGE);
-		}
+		JFrame frame = new JFrame(ScaleTurtlePanel.class.getSimpleName());
+		frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
```

Comment:
EXIT_ON_CLOSE causes system.exit() to be called, as discussed elsewhere.  DISPOSE_ON_CLOSE might be what you're looking for.

Reply:
the purpose of this modification is to remove the `catch` and let the exception be handled by the `main()` method which will display it in the console and exit in error if something wrong happen. Nothing related to `EXIT_ON_CLOSE` or else ;-)